cpp/ql/src/experimental/Security/CWE/CWE-020/LateCheckOfFunctionArgument.ql

```diff
@@ -0,0 +1,66 @@
+/**
+ * @name Late Check Of Function Argument
+ * @description --Checking the function argument after calling the function itself.
+ *              --This situation looks suspicious and requires the attention of the developer.
+ *              --It may be necessary to add validation before calling the function.
+ * @kind problem
+ * @id cpp/late-check-of-function-argument
+ * @problem.severity warning
+ * @precision medium
+ * @tags correctness
+ *       security
+ *       external/cwe/cwe-20
+ */
+
+import cpp
+import semmle.code.cpp.valuenumbering.GlobalValueNumbering
+
+predicate numberArgument(Function f, int size) {
```

Comment:
Could we have a comment explaining what this predicate is?  I thought `size` was the index of a parameter to `f` that is the size of a buffer (as in for `write` and `read`), but that isn't true for `lseek`.  It would be good if future maintainers can add to this predicate with confidence that they've understood it correctly.

Reply:
this predicate allows you to get the number of the argument, which has an impact on the positioning in the buffer. the list of functions at this stage is generated in manual mode.